File: Backend/model/model_user.py
from sqlmodel import SQLModel, Field , select ,Session
from database import get_session
from fastapi import Depends
from pydantic import BaseModel
from typing import Optional, List
from sqlalchemy import Column, JSON

class User(SQLModel, table=True):
    id: int = Field(default=None, primary_key=True)
    nik: int
    nama: str
    email: str
    password: str
    face_encoding: Optional[List[float]] = Field(
        default=None,
        sa_column=Column(JSON)
    )


class LoginInput(BaseModel):
    nik: int
    password: str

def find_user(nik: int, session: Session): 
    statement = select(User).where(User.nik == nik)
    return session.exec(statement).first()

File: Frontend/lib/model/model_user.dart
class UserModelRegister {
  final int nik;
  final String nama;
  final String email;
  final String password;

  UserModelRegister({
    required this.nik,
    required this.nama,
    required this.email,
    required this.password,
  });

  factory UserModelRegister.fromJson(Map<String, dynamic> json) {
    return UserModelRegister(
      nik: json['nik'],
      nama: json['nama'],
      email: json['email'],
      password: json['password'],
    );
  }

  Map<String, dynamic> toJson() {
    return {"nik": nik, "nama": nama, "email": email, "password": password};
  }
}

class UserModelLogin {
  final int nik;
  final String password;

  UserModelLogin({required this.nik, required this.password});

  factory UserModelLogin.fromJson(Map<String, dynamic> json) {
    return UserModelLogin(nik: json['nik'], password: json['password']);
  }

  Map<String, dynamic> toJson() {
    return {"nik": nik, "password": password};
  }
}
